File: 661.py
import numpy as np

class Solution:
    def imageSmoother(self, img: List[List[int]]) -> List[List[int]]:
        m, n = len(img), len(img[0])
        imgnp_pad = np.zeros((m + 2, n + 2), dtype=np.uint16)
        imgnp_pad[1:-1, 1:-1] = np.asarray(img, dtype=np.uint16)
        counter_pad = np.zeros((m + 2, n + 2), dtype=np.uint16)
        counter_pad[1:-1, 1:-1] = 1
        imgnp = np.zeros((m, n), dtype=np.uint16)
        counter = np.zeros((m, n), dtype=np.uint16)
        for i in range(3):
            for j in range(3):
                imgnp += imgnp_pad[i:i + m, j:j + n]
                counter += counter_pad[i:i + m, j:j + n]
        return (imgnp // counter).tolist()

File: sol/661/661.cpp
struct Solution {
    vector<vector<int>> imageSmoother(vector<vector<int>>& img) {
        int m = img.size(), n = img[0].size(), counter, total;
        vector<vector<int>> newimg(m, vector<int>(n, 0));
        for (int i = 0; i < m; ++i){
            for (int j = 0; j < n; ++j){
                counter = 0;
                total = 0;
                for (int di = -1; di <= 1; ++di){
                    for (int dj = -1; dj <= 1; ++dj){
                        int ii = i + di, jj = j + dj;
                        if (!(0 <= ii && ii < m && 0 <= jj && jj < n)){
                            continue;
                        }
                        total += img[ii][jj];
                        ++counter;
                    }
                }
                newimg[i][j] = total / counter;
            }
        }
        return newimg;
    }
};
